backend/routes/generate.js

```diff
@@ -0,0 +1,290 @@
+function avoidTimesConstraint(possibleSchedules, avoidTimes){
+    for (var i=0; i<possibleSchedules.length; i++) {
+        schedule = possibleSchedules[i];
+
+        for (var classData of schedule) {
+            // console.log("classData iteration to detect conflict with avoid times")
+            // check conflict with avoid time list
+            // if conflict remove schedule from possibleSchedules
+            if (conflictAvoidTime(avoidTimes, classData)) {
+                // console.log("removed schedule")
+                index = possibleSchedules.indexOf(schedule);
+                if (index > -1) {
+                    // console.log(possibleSchedules[index])
+                    possibleSchedules.splice(index, 1);
+                    i-=1;
+                }    
+                break;
+            }
+        } 
+    }
+    return possibleSchedules
+}
+
+function sectionConstraint(possibleSchedules, avoidTimes){
+    result = [];
+    for (var i=0; i<possibleSchedules.length; i++) {
+        schedule = possibleSchedules[i];
+
+        for (var classData of schedule) {
+            classCopy = classData;
+            if (classData.sections) {
+                for (var section of classData.sections) {
+                    // remove from class.sections array if conflict with schedule
+                    // or avoid time list
+                    if (checkTimeConflict(schedule, section) || conflictAvoidTime(avoidTimes, section)) {
+                        classCopy.sections.splice(classData.sections.indexOf(section), 1);
+                    }
+                }
+            }
+            // console.log(classCopy);
+            // store the new class information
+            schedule[schedule.indexOf(classData)] = classCopy;
+        } 
+        result.push(schedule)
+    }
+    return result
+}
+
+function priorityReorder(possibleSchedules) {
+    priorityScores = [];
+    reorderedList = [];
+    // get priorityScores for each schedule
+    for (var i=0; i<possibleSchedules.length; i++) {
+        schedule = possibleSchedules[i];
+        score = 0;
+        for (var classData of schedule) {
+            score += classData.priority;
+        }
+        priorityScores.push(score);
+    }
+    // console.log("priority score:", priorityScores);
+
+    // priority reorder based on priority score
+    // the schedule with minimum score means higher priority
+    // will be reordered and displayed in the front of the list
+    priorityScoresCopy = [...priorityScores];
+    while (priorityScoresCopy && priorityScoresCopy.length) {
+        min = Infinity;
+        for (var score of priorityScoresCopy) {
+            if (score < min) {
+                min = score;
+            }
+        }
+        priorityScoresCopy.splice(priorityScoresCopy.indexOf(min), 1);
+        reorderedList.push(possibleSchedules[priorityScores.indexOf(min)]);
+    }
+
+    return reorderedList
+}
+
+
+function generateSchedules(requestObject) {
```

Comment:
probably should move this function to the very top or very bottom since it's the most parent function